connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerTask.java

```diff
@@ -231,4 +248,202 @@ public void transitionTo(TargetState state) {
         }
     }
 
-}
+    /**
+     * Record that offsets have been committed.
+     *
+     * @param duration the length of time in milliseconds for the commit attempt to complete
+     */
+    protected void recordCommitSuccess(long duration) {
+        taskMetricsGroup.recordCommit(duration, true, null);
+    }
+
+    /**
+     * Record that offsets have been committed.
+     *
+     * @param duration the length of time in milliseconds for the commit attempt to complete
+     * @param error the unexpected error that occurred; may be null in the case of timeouts or interruptions
+     */
+    protected void recordCommitFailure(long duration, Throwable error) {
+        taskMetricsGroup.recordCommit(duration, false, error);
+    }
+
+    /**
+     * Record that a batch of records has been processed.
+     *
+     * @param size the number of records in the batch
+     */
+    protected void recordBatch(int size) {
+        taskMetricsGroup.recordBatch(size);
+    }
+
+    TaskMetricsGroup taskMetricsGroup() {
+        return taskMetricsGroup;
+    }
+
+    static class TaskMetricsGroup implements TaskStatus.Listener {
+        private final TaskStatus.Listener delegateListener;
+        private final MetricGroup metricGroup;
+        private final Time time;
+        private final StateTracker taskStateTimer;
+        private final Sensor commitTime;
+        private final Sensor batchSize;
+        private final Sensor commitAttempts;
+
+        public TaskMetricsGroup(ConnectorTaskId id, ConnectMetrics connectMetrics, TaskStatus.Listener statusListener) {
+            delegateListener = statusListener;
+            time = connectMetrics.time();
+            taskStateTimer = new StateTracker();
+            metricGroup = connectMetrics.group("connector-tasks",
+                    "connector", id.connector(), "task", Integer.toString(id.task()));
+
+            addTaskStateMetric(State.UNASSIGNED, "status-unassigned",
+                    "Signals whether the connector task is in the unassigned state.");
+            addTaskStateMetric(State.RUNNING, "status-running",
+                    "Signals whether the connector task is in the running state.");
+            addTaskStateMetric(State.PAUSED, "status-paused",
+                    "Signals whether the connector task is in the paused state.");
+            addTaskStateMetric(State.FAILED, "status-failed",
+                    "Signals whether the connector task is in the failed state.");
+            addTaskStateMetric(State.DESTROYED, "status-destroyed",
+                    "Signals whether the connector task is in the destroyed state.");
+
+            addRatioMetric(State.RUNNING, "running-ratio",
```

Comment:
this state is missing from the KIP, it should be added

Reply:
Ah, good catch. Fixed.